src/main/java/st/ratpack/auth/CachingTokenValidator.java

```diff
@@ -0,0 +1,37 @@
+package st.ratpack.auth;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import ratpack.exec.Promise;
+
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+public class CachingTokenValidator implements TokenValidator {
+
+	public static long DEFAULT_TTL = 5000;
+
+	private final LoadingCache<String, Promise<Optional<OAuthToken>>> cache;
+
+	public CachingTokenValidator(AuthModule.Config config, final TokenValidator tokenValidator) {
+		Long ttl = config.ttl != null ? config.ttl : DEFAULT_TTL;
+		System.out.println(tokenValidator);
+
+		cache = CacheBuilder.newBuilder()
+				.expireAfterWrite(ttl, TimeUnit.MILLISECONDS)
+				.build(new CacheLoader<String, Promise<Optional<OAuthToken>>>() {
```

Comment:
You were correct that we wanted to cache the promise its more performant that way but we want to make sure we cache a Value promise not the raw promise from other calls.